iceoryx_hoofs/test/moduletests/test_posix_periodic_timer.cpp

```diff
@@ -0,0 +1,225 @@
+// Copyright (c) 2021 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_hoofs/error_handling/error_handling.hpp"
+#include "iceoryx_hoofs/internal/units/duration.hpp"
+#include "iceoryx_hoofs/posix_wrapper/periodic_timer.hpp"
+#include "iceoryx_hoofs/testing/timing_test.hpp"
+#include "test.hpp"
+
+#include <chrono>
+#include <thread>
+
+namespace
+{
+using namespace ::testing;
+using namespace iox::units;
+using namespace iox::units::duration_literals;
+
+using Timer = iox::posix::PeriodicTimer;
+
+class PeriodicTimer_test : public Test
+{
+  public:
+    virtual void SetUp()
+    {
+        numberOfCalls = 0;
+    }
+
+    virtual void TearDown()
+    {
+    }
+
+    Duration second{1_s};
+
+    std::atomic<int> numberOfCalls{0};
+    static const Duration INTERVAL;
+};
+
+const Duration PeriodicTimer_test::INTERVAL{50_ms};
+
+TEST_F(PeriodicTimer_test, TimerAutoStartTest)
+{
+    Timer sut(0_s);
+
+    auto timerState = sut.wait(iox::posix::TimerCatchupPolicy::IMMEDIATE_TICK);
+
+    ASSERT_FALSE(timerState.has_error());
```

Comment:
Please use `ASSERT` for cases that the test cannot proceed if this is not fulfilled. For instance a case like:
```cpp
ASSERT_FALSE(vector.empty());
vector[0]; // would lead to segfault if vector would be empty.
```

Reply:
Yes. modified to check if the expected has errors.